lib/device.c

```diff
@@ -295,10 +316,46 @@ static int read_track_data(struct sync_device *d, struct sync_track *t)
 	return 0;
 }
 
+static int create_leading_dirs(const char *path)
+{
+	char *pos, buf[FILENAME_MAX];
+
+	strncpy(buf, path, sizeof(buf));
+	buf[sizeof(buf) - 1] = '\0';
+	pos = buf + (*buf == '/');
+
+	while (1) {
+		struct stat st;
+
+		pos = strchr(pos, '/');
+		if (!pos)
+			break;
+		*pos = '\0';
+
+		/* does path exist, but isn't a dir? */
+		if (!stat(buf, &st)) {
+			if (!S_ISDIR(st.st_mode))
+				return -1;
+		} else {
+			if (mkdir(buf, 0777))
```

Comment:
maybe mode 0644 (write+read owner and read for everyone else, no execution) would make more sense as the default?

Reply:
Yeah, that's true. Thanks for pointing that out :)

Reply:
Actually, 0644 doesn't seem to work, 0744 seems to do the trick.